### PR TITLE
Fixing permission for admin wizard

### LINKED ISSUE #6060: [CAL-606] Admin Wizard visible for non-admin users
### Issue Summary

Admin Wizard visible for non-admin users.

> **Note**
> This is not an urgent matter, the admin creation on step 1 is already disabled when there are users in the DB, and the second step to enable apps does not return any app for non-admin users

### Steps to Reproduce

1. Log in as a regular user in any environment
2. Go to `/auth/setup` and see the admin wizard
3. Go to `/auth/setup?step=2` and see the app selection (not returning options luckily)

<sub>[CAL-606](https://linear.app/calcom/issue/CAL-606/admin-wizard-visible-for-non-admin-users)</sub>

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Adding a redirection to 400 error when logged in user's role is not ADMIN.

Fixes #6060 

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

1. Log in as a regular user and go to `/auth/setup` and you will get redirected to the 400 error

### CODE REVIEW

File: apps/web/pages/auth/setup/index.tsx [hunk 1]
```diff
@@ -38,8 +41,22 @@ export default function Setup(props: inferSSRProps<typeof getServerSideProps>) {
   );
 }
 
-export const getServerSideProps = async () => {
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
   const userCount = await prisma.user.count();
+  const { req } = context;
+  const session = await getSession({ req });
+
+  console.log({ role: session?.user.role });
```
Comment (hariombalhara) [inline apps/web/pages/auth/setup/index.tsx:49]: nit: remove console.log

File: apps/web/pages/auth/setup/index.tsx [hunk 2]
```diff
@@ -38,8 +41,22 @@ export default function Setup(props: inferSSRProps<typeof getServerSideProps>) {
...
+
+  if (session?.user.role && session?.user.role !== UserPermissionRole.ADMIN) {
+    return {
+      redirect: {
+        destination: `/400`,
```
Comment (hariombalhara) [inline apps/web/pages/auth/setup/index.tsx:54]: 403 or 404 seems better option to me.
403 - Clearly tells that you are not authorized
404 - We want to hide that the user is accessing a valid route

**Review by hariombalhara — APPROVED** [review]

File: apps/web/pages/auth/setup/index.tsx (hunk 1 above)
Comment (leog) [inline apps/web/pages/auth/setup/index.tsx:49]:
```suggestion
```

File: apps/web/pages/auth/setup/index.tsx (hunk 2 above)
Comment (leog) [reply apps/web/pages/auth/setup/index.tsx:53]: @hariombalhara I think we only have error 400 and 500 at the moment

Comment (zomars) [reply apps/web/pages/auth/setup/index.tsx:53]: @leog we don't have a 400 error page tho

Comment (leog) [reply apps/web/pages/auth/setup/index.tsx]: My bad! 😅

**Review by PeerRich — APPROVED** [review]
LGTM

**Review by JeroenReumkens — APPROVED** [review]

**Pull request merged**